### PR TITLE
PYIC-3514 Refactor code to separate audit service as a separate dependent service. Also separate common services.

### PR DESCRIPTION
**Pull request by rpayal**
## Proposed changes
Refactor Audit service to separate lib
Refactor to extract common code from lib and move to separate common service.

### Issue tracking

- [PYIC-3514](https://govukverify.atlassian.net/browse/PYIC-3514)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[PYIC-3514]: https://govukverify.atlassian.net/browse/PYIC-3514?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: deploy/template.yaml [hunk 1]
```diff
@@ -95,12 +95,12 @@ Mappings:
       journeyEngineStepFunctionLogLevel: "ALL"
       pgw500ErrorLimit: 2
       egw500ErrorLimit: 2
-      asyncCriResponseQueueArn: "arn:aws:sqs:eu-west-2:616199614141:stubQueue_F2FQueueOldDev"
+      asyncCriResponseQueueArn: "arn:aws:sqs:eu-west-2:616199614141:stubQueue_F2FQueue"
```
Comment (Wynndow) [inline deploy/template.yaml]: This change seems unrelated - did you mean to include it?

File: libs/audit-service/build.gradle [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	id 'java-library'
+	id "idea"
+	id "jacoco"
+}
+
+repositories {
+	mavenCentral()
+}
+
+def var = dependencies {
```
Comment (Wynndow) [inline libs/audit-service/build.gradle]: What does declaring `var` give you?

File: libs/common-services/src/test/java/uk/gov/di/ipv/core/library/domain/gpg45/Gpg45ProfileEvaluatorTest.java [hunk 3]
```diff
@@ -343,7 +344,8 @@ void getCredentialByTypeShouldReturnCorrectType() throws Exception {
 
         JWTClaimsSet jwtClaimsSet = result.get().getJWTClaimsSet();
 
-        assertEquals("https://review-p.integration.account.gov.uk", jwtClaimsSet.getIssuer());
+        Assertions.assertEquals(
```
Comment (Wynndow) [inline libs/common-services/src/test/java/uk/gov/di/ipv/core/library/domain/gpg45/Gpg45ProfileEvaluatorTest.java]: I think we can just use the static import we've already got, so just `assertEquals()`

File: libs/common-services/src/test/java/uk/gov/di/ipv/core/library/domain/gpg45/Gpg45ProfileEvaluatorTest.java [hunk 4]
```diff
@@ -359,7 +361,8 @@ void getCredentialByTypeShouldReturnCorrectTypeForDcmaw() throws Exception {
 
         JWTClaimsSet jwtClaimsSet = result.get().getJWTClaimsSet();
 
-        assertEquals("https://review-b.integration.account.gov.uk", jwtClaimsSet.getIssuer());
+        Assertions.assertEquals(
```
Comment (Wynndow) [inline libs/common-services/src/test/java/uk/gov/di/ipv/core/library/domain/gpg45/Gpg45ProfileEvaluatorTest.java]: Same here

File: libs/common-services/src/test/java/uk/gov/di/ipv/core/library/domain/gpg45/Gpg45ProfileEvaluatorTest.java [hunk 5]
```diff
@@ -380,7 +383,7 @@ void getCredentialByTypeShouldReturnCorrectTypeForF2F() throws Exception {
 
         JWTClaimsSet jwtClaimsSet = result.get().getJWTClaimsSet();
 
-        assertEquals(
+        Assertions.assertEquals(
```
Comment (Wynndow) [inline libs/common-services/src/test/java/uk/gov/di/ipv/core/library/domain/gpg45/Gpg45ProfileEvaluatorTest.java]: And here

File: libs/common-services/src/test/java/uk/gov/di/ipv/core/library/helpers/APIGatewayResponseGeneratorTest.java [hunk 6]
```diff
@@ -17,14 +17,14 @@ void proxyJsonResponseRaises500IfInvalidJson() {
...
     }
 
     @Test
     void proxyJsonResponseReturnsValidResponse() throws JsonProcessingException {
         APIGatewayProxyResponseEvent response =
                 ApiGatewayResponseGenerator.proxyJsonResponse(200, "test");
-        assertEquals(200, response.getStatusCode());
-        assertEquals(new ObjectMapper().writeValueAsString("test"), response.getBody());
+        Assertions.assertEquals(200, response.getStatusCode());
+        Assertions.assertEquals(new ObjectMapper().writeValueAsString("test"), response.getBody());
```
Comment (Wynndow) [inline libs/common-services/src/test/java/uk/gov/di/ipv/core/library/helpers/APIGatewayResponseGeneratorTest.java]: And these

**Review by Wynndow — COMMENTED** [review]
Needs a rebase, and a few import things, but looks good!

File: deploy/template.yaml (hunk 1 above)
Comment (rpayal) [reply deploy/template.yaml]: thats Vam's change.

File: libs/audit-service/build.gradle (hunk 2 above)
Comment (rpayal) [reply libs/audit-service/build.gradle]: trying to remember....added before to resolve issue with gradle build. Removed now.

File: libs/common-services/src/test/java/uk/gov/di/ipv/core/library/helpers/APIGatewayResponseGeneratorTest.java (hunk 6 above)
Comment (rpayal) [reply libs/common-services/src/test/java/uk/gov/di/ipv/core/library/helpers/APIGatewayResponseGeneratorTest.java]: done

File: libs/common-services/src/test/java/uk/gov/di/ipv/core/library/domain/gpg45/Gpg45ProfileEvaluatorTest.java (hunk 5 above)
Comment (rpayal) [reply libs/common-services/src/test/java/uk/gov/di/ipv/core/library/domain/gpg45/Gpg45ProfileEvaluatorTest.java]: done

File: libs/common-services/src/test/java/uk/gov/di/ipv/core/library/domain/gpg45/Gpg45ProfileEvaluatorTest.java (hunk 4 above)
Comment (rpayal) [reply libs/common-services/src/test/java/uk/gov/di/ipv/core/library/domain/gpg45/Gpg45ProfileEvaluatorTest.java]: done

File: libs/common-services/src/test/java/uk/gov/di/ipv/core/library/domain/gpg45/Gpg45ProfileEvaluatorTest.java (hunk 3 above)
Comment (rpayal) [reply libs/common-services/src/test/java/uk/gov/di/ipv/core/library/domain/gpg45/Gpg45ProfileEvaluatorTest.java]: done

**Review by Wynndow — APPROVED** [review]
